### ISSUE TITLE
Cannot make an input of HashMap<String, ..> with quoted strings

### ISSUE THREAD

**Issue by Visic** [async-graphql/async-graphql#500]
## Expected Behavior
Input values of the type "HashMap<String, Json<serde_json::Value>>" should allow quoted strings for the keys.

## Actual Behavior
The input type is turned into a JSONObject! which seems to want valid graphql "names", but those are restricted to alphanumeric characters and underscores.

**Comment by sunli829** [comment]
I don't think this is a bug, quoted string for the key does not conform to the GraphQL spec.

**Comment by Visic** [comment]
I believe you are correct, if this crate offers support for HashMap<String, ...> as an input value, what it is doing is the most reasonable. I suppose my disconnect was that when I wrote my input as the type HashMap<String, ...> I expected (and needed) to allow for any string, so when I went to use it and it didn't work that way, I was surprised.

Ultimately, I think the only "fix" would be to not support the HashMap<String, ...> as an input type and require the use of the JSONObject type instead, which may make this less of a footgun. But seeing as that would be a breaking change, and it wouldn't achieve my original intent, I am content with closing this issue.
